### ISSUE TITLE
Import resolves to `undefined`

### ISSUE THREAD

**Issue by laymonage** [timlrx/rehype-prism-plus#12]
Hey, thanks for the plugin.

I've been trying to update my blog to use ESM for the MDX stuff (mostly remark and rehype plugins). Everything seems to work, except this plugin. When loading the page, Next.js throws this obscure error:

![image](https://user-images.githubusercontent.com/6379424/132377744-ae2246c3-fc6f-4ec1-a79c-cc5740521df9.png)

In the terminal, the following error shows up:

![image](https://user-images.githubusercontent.com/6379424/132377880-68035c80-56c2-46a7-a00d-efa61595b5b9.png)

Upon further inspection, it seems that the import for `rehype-prism-plus` resolves to `undefined`.

```ts
import rehypePrismPlus from 'rehype-prism-plus';
// ...
// DEBUG
console.log({ rehypePrismPlus });
```

![image](https://user-images.githubusercontent.com/6379424/132378070-21f10fe4-f7ee-4fad-8d9b-210ca815fb44.png)

However, if I try to `console.log({ rehypePrismPlus });` in client-side code, it seems to resolve correctly:

![image](https://user-images.githubusercontent.com/6379424/132379010-790746db-2df4-48b4-9a01-ed360b1e8796.png)

I have no idea what's causing the issue. If I remove `rehypePrismPlus` from `options.rehypePlugins`, everything works fine.

This issue doesn't show up with your blog template, though I can't seem to find out any difference in my blog that may cause this.

I know that it's probably just an issue with my setup, and cloning my whole blog is probably too much. However, if you'd like to look into it, I'd really appreciate it. You can check out the branch of this PR: laymonage/base#35.

Thanks!

**Comment by timlrx** [comment]
Hi, I traced the issue down to the `withPreact` plugin. After removing `withPreact` and using installing `react`, I managed to import the module properly. I have preact working with my setup by manually specifying the aliases. Not sure what else is added by `withPreact` and if there's something I could fix on my side. It seems to work with the other remark rehype esm packages.

But at least we found the cause for this rather perplexing bug!

**Comment by laymonage** [comment]
Thanks @timlrx! I'll try to remove the Preact plugin. That was my last suspicion before I post the issue. I did notice that we had different setups for hooking Preact, but as the other plugins are working, I thought it might be unrelated. If I get it working, I will close this issue.

However, if you somehow found the fix from your side, that'd be great too! The Preact plugin is just very convenient and I'd rather use it if possible :smile:

**Comment by timlrx** [comment]
I suspect it might be related to this: https://github.com/preactjs/next-plugin-preact/blob/master/packages/next-plugin-preact/index.js#L23-L28

I publish both the cjs and esm packages. Maybe you can try deleting the cjs from the node modules and see whether you can get it working that way? If it works, I will publish an esm only package like most of the current updated remark / rehype packages.

**Comment by laymonage** [comment]
@timlrx I tried removing the `.umd.js` file and its mentions in the `package.json`. Nope, it still shows the same error :(

**Comment by timlrx** [comment]
Actually, that works for me. I think you might have forgotten to change the `main` to the `es.js` one instead of just deleting it. 

In any case, I published a v1.0.0 version which is ESM only and it works on your blog :)

**Comment by laymonage** [comment]
Pretty sure I did, but no dice:
(I've also tried restarting the dev server).

![image](https://user-images.githubusercontent.com/6379424/132544738-0caf3a77-ae76-4077-9f94-521ff1e6cd9b.png)

Anyway, v1.0.0 works flawlessly. Thank you very much! :partying_face:
